packages/terra-clinical-result/src/ClinicalResult.jsx

```diff
@@ -0,0 +1,116 @@
+   */
+  hideUnit: PropTypes.bool,
+  /**
+   * Whether or not the text should be truncated in display. Restricts clinical result details each to one line.
+   */
+  isTruncated: PropTypes.bool,
+  /**
+   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
+   */
+  hasResultError: PropTypes.bool,
+  /**
+   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
+   */
+  hasResultNoData: PropTypes.bool,
+};
+
+const defaultProps = {
+  resultData: {},
+  hideUnit: false,
+  isTruncated: false,
+  hasResultError: false,
+  hasResultNoData: false,
+};
+
+const ClinicalResult = (props) => {
+  const {
+    resultData,
+    hideUnit,
+    isTruncated,
+    hasResultError,
+    hasResultNoData,
+    ...customProps
+  } = props;
+
+  let clinicalResultDisplay = null;
+
+  if (hasResultError || hasResultNoData) {
+    clinicalResultDisplay = hasResultError ? (<ResultError />) : (<NoData />);
+  } else {
+    const modifiedIconElement = resultData.isModified && !resultData.isUnverified ? (<IconModified className={cx('icon-modified')} />) : null;
+    const commentIconElement = resultData.hasComment && !resultData.isUnverified ? (<IconComment className={cx('icon-comment')} />) : null;
+    const unverifiedIconElement = resultData.isUnverified ? (<IconUnverified className={cx('icon-unverified')} />) : null;
+    const iconGroupModifiedComment = resultData.isModified || resultData.hasComment || resultData.isUnverified ? (
+      <React.Fragment>
+        {modifiedIconElement}
+        {commentIconElement}
+        {unverifiedIconElement}
+      </React.Fragment>
+    ) : null;
+
+    const conceptDisplayElement = resultData.conceptDisplay ? (<div className={cx('concept-display')}>{resultData.conceptDisplay}</div>) : null;
+    const datetimeDisplayElement = resultData.datetimeDisplay ? (<div className={cx('datetime-display')}>{resultData.datetimeDisplay}</div>) : null;
+
+    const decoratedResultClassnames = cx([
+      'decorated-result-display',
+      { truncated: isTruncated },
+    ]);
+
+    clinicalResultDisplay = (
+      <React.Fragment>
+        <div className={decoratedResultClassnames}>
+          <div className={cx('result-display')}>
+            <Observation
+              eventId={resultData.eventId}
+              result={resultData.result}
+              interpretation={resultData.interpretation}
+              isUnverified={resultData.isUnverified}
+              hideUnit={hideUnit}
+            />
+            {isTruncated ? null : iconGroupModifiedComment}
+          </div>
+          {isTruncated ? iconGroupModifiedComment : null}
+        </div>
+        {conceptDisplayElement}
+        {datetimeDisplayElement}
+      </React.Fragment>
+    );
+  }
+
+  const clinicalResultClassnames = cx([
+    'clinical-result',
+    customProps.className,
```

Comment:
You shouldn't run custom props through the CX function.
```javascript
  const clinicalResultClassnames = [
    cx('clinical-result'),
    customProps.className,
   ].join(' ');
```

Reply:
This is/was a wide-spread used pattern in other components, might not hurt to put it out as a general update to make to all other components are they are updated and/or put into our linter.

Resolved here: https://github.com/cerner/terra-clinical/pull/574/commits/6f0e60f1a7d9dd7530464b1470d3585a02b6e30e

instead using same pattern as [../terra-table/src/Table.jsx#L409](https://github.com/cerner/terra-core/blob/4b4584b518192c68e059a4bcd5e1b4160b1e0807/packages/terra-table/src/Table.jsx#L409) to avoid adding extra joined space char.
```
<div
  {...customProps}
  className={customProps.className ? `${tableClasses} ${customProps.className}` : tableClasses}
>
```